axum/src/docs/middleware.md

```diff
@@ -61,12 +61,12 @@ let app = Router::new()
 
 Some commonly used middleware are:
 
-- [`TraceLayer`](tower_http::trace) for high level tracing/logging.
-- [`CorsLayer`](tower_http::cors) for handling CORS.
-- [`CompressionLayer`](tower_http::compression) for automatic compression of
+- [`TraceLayer`](tower_http::trace::TraceLayer) for high level tracing/logging.
```

Comment:
I don't see why intra-doc links would work for the type, but not the module. In general I wouldn't really have expected this to work at all, they're called **intra**-doc links afterall. @davidpdrsn are you sure this used to work?

Reply:
- \- \[\`TraceLayer\`\](tower_http::trace) works fine locally, but somehow on [docs.rs](https://docs.rs/axum/latest/axum/middleware/) it's broken.
- Also, earlier I thought the issue was with linking the whole module rather than the specific Layer.

Reply:
I see. Can you try this:

* Make sure you have an up-to-date nightly toolchain
* Create `.cargo/config.toml` with the following content:
  ```toml
  [doc.extern-map.registries]
  crates-io = "https://docs.rs/"
  
  [unstable]
  rustdoc-map = true
  ```
* Run `RUSTDOCFLAGS='--cfg docsrs' cargo +nightly doc --no-deps --all-features`

That should get you very close to how docs.rs generates documentation, maybe you can reproduce the issue then.

Reply:
Following the steps specified, both branches `upstream/main` and `origin/brokendocslink` works fine. Former linking the whole module, while latter the specific layer.

```
rustc 1.64.0-nightly (f8588549c 2022-07-18)
binary: rustc
commit-hash: f8588549c3c3d45c32b404210cada01e2a45def3
commit-date: 2022-07-18
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

Reply:
Interesting. I wonder whether this was just a nightly regression then that got fixed since we last published a release on docs.rs, or some way in which docs.rs does things differently to that setup.

Reply:
Should I try it on the released doc's nightly version? But I don't know how to get that.

Reply:
Just tried shipping a new version and thus rebuilding the docs. [But this issue is still there.](https://github.com/tokio-rs/axum/issues/1180#issuecomment-1194327015). So seems we do need this after all. Not clue why it works locally but nto on docs.rs

Reply:
These links shouldn't go to the `Layer` but the module, so `tower_http::trace` instead of `tower_http::trace::TraceLayer`. The modules contains the docs an examples.

Also seems we need to use absolute links rather than intra docs links. Not sure why their broken on docs.rs 🤷

Reply:
Does absolute links here means direct link to `docs.rs`?

Reply:
Yes.